core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -277,6 +294,245 @@ impl SanitizedTransactionReceiveAndBuffer {
     }
 }
 
+pub(crate) struct TransactionViewReceiveAndBuffer {
+    pub receiver: BankingPacketReceiver,
+    pub bank_forks: Arc<RwLock<BankForks>>,
+}
+
+impl ReceiveAndBuffer for TransactionViewReceiveAndBuffer {
+    type Transaction = RuntimeTransaction<ResolvedTransactionView<Bytes>>;
+    type Container = TransactionViewStateContainer;
+
+    fn receive_and_buffer_packets(
+        &mut self,
+        container: &mut Self::Container,
+        timing_metrics: &mut SchedulerTimingMetrics,
+        count_metrics: &mut SchedulerCountMetrics,
+        decision: &BufferedPacketsDecision,
+    ) -> bool {
+        let (root_bank, working_bank) = {
+            let bank_forks = self.bank_forks.read().unwrap();
+            let root_bank = bank_forks.root_bank();
+            let working_bank = bank_forks.working_bank();
+            (root_bank, working_bank)
+        };
+
+        // Receive packet batches.
+        const TIMEOUT: Duration = Duration::from_millis(10);
+        let start = Instant::now();
+        let mut received_message = false;
+
+        // If not leader/unknown, do a blocking-receive initially. This lets
+        // the thread sleep until a message is received, or until the timeout.
+        // Additionally, only sleep if the container is empty.
+        if container.is_empty()
+            && matches!(
+                decision,
+                BufferedPacketsDecision::Forward | BufferedPacketsDecision::ForwardAndHold
```

Comment:
removed `Hold` branch here. Sometimes after a leader slot, we do not have a bank ready in `Consume` for our next leader slot. We should still avoid the `recv_timeout` in that case.